code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm

```diff
@@ -540,3 +544,207 @@
+	metabolization_rate = 0.01 * REM
+	chemical_flags = REAGENT_DEAD_PROCESS
+	tox_damage = 0
+	///The old heart we're swapping for
+	var/obj/item/organ/heart/original_heart
+	///The new heart that's temp added
+	var/obj/item/organ/heart/cursed/manual_heart
+
+///Creates a new cursed heart and puts the old inside of it, then replaces the position of the old
+/datum/reagent/inverse/corazone/on_mob_add(mob/living/owner)
+	if(!iscarbon(owner))
+		return
+	var/mob/living/carbon/carbon_mob = owner
+	original_heart = owner.getorganslot(ORGAN_SLOT_HEART)
+	if(!original_heart)
+		return
+	manual_heart = new(null, src)
+	original_heart.Remove(carbon_mob, special = TRUE) //So we don't suddenly die
+	original_heart.forceMove(manual_heart)
+	original_heart.organ_flags |= ORGAN_FROZEN //Not actually frozen, but we want to pause decay
+	manual_heart.Insert(carbon_mob, special = TRUE)
+	//these last so instert doesn't call them
+	RegisterSignal(carbon_mob, COMSIG_CARBON_GAIN_ORGAN, .proc/on_gained_organ)
+	RegisterSignal(carbon_mob, COMSIG_CARBON_LOSE_ORGAN, .proc/on_removed_organ)
+	to_chat(owner, "<span class='warning'>You feel your heart suddenly stop beating on it's own - you'll have to manually beat it!</spans>")
+	..()
+
+///Intercepts the new heart and creates a new cursed heart - putting the old inside of it
+/datum/reagent/inverse/corazone/proc/on_gained_organ(mob/owner, obj/item/organ/organ)
+	SIGNAL_HANDLER
+	if(!istype(organ, /obj/item/organ/heart))
+		return
+	var/mob/living/carbon/carbon_mob = owner
+	original_heart = organ
+	original_heart.Remove(carbon_mob, special = TRUE)
+	original_heart.forceMove(manual_heart)
+	original_heart.organ_flags |= ORGAN_FROZEN //Not actually frozen, but we want to pause decay
+	if(!manual_heart)
+		manual_heart = new(null, src)
+	manual_heart.Insert(carbon_mob, special = TRUE)
+
+///If we're ejecting out the organ - replace it with the original
+/datum/reagent/inverse/corazone/proc/on_removed_organ(mob/prev_owner, obj/item/organ/organ)
+	SIGNAL_HANDLER
+	if(!organ == manual_heart)
+		return
+	original_heart.forceMove(organ.loc)
+	original_heart.organ_flags &= ~ORGAN_FROZEN //enable decay again
+	organ.forceMove(null)
+	qdel(organ)
+
+///We're done - remove the curse and restore the old one
+/datum/reagent/inverse/corazone/on_mob_delete(mob/living/owner)
+	//Do these first so Insert doesn't call them
+	UnregisterSignal(owner, COMSIG_CARBON_LOSE_ORGAN)
+	UnregisterSignal(owner, COMSIG_CARBON_GAIN_ORGAN)
+	if(!iscarbon(owner))
+		return
+	var/mob/living/carbon/carbon_mob = owner
+	if(original_heart) //Mostly a just in case
+		original_heart.organ_flags &= ~ORGAN_FROZEN //enable decay again
+		original_heart.Insert(carbon_mob, special = TRUE)
+	manual_heart.forceMove(null) //so we can be sure this is removed
+	qdel(manual_heart)
+	to_chat(owner, "<span class='warning'>You feel your heart start beating normally again!</spans>")
+	..()
+
+/datum/reagent/inverse/antihol
```

Comment:
Done!

Reply:
>>> and what if the mob's heart is currently undergoing heart failure? I guess since they're manually beating it, it sort of makes sense that their heart could still work...

Hmm, I'm kind of okay with it overriding the failure as a sort of inpromptu treatment for it - it's not like the old heart disappears, so it should come back after it's done!

Reply:
> also also, check how this interacts with abductor glands, they could get kinda funky

So looking over glands, it looks like I might have to write some snowflake code, hmm.